test/test_network.h

```diff
@@ -137,17 +137,17 @@ TEST(network, manual_init) {
 
   adagrad opt;
 
-  vec_t *c1_w = net[0]->weights()[0];
-  vec_t *c1_b = net[0]->weights()[1];
-  vec_t *f1_w = net[1]->weights()[0];
+  Parameter c1_w = net[0]->parameter_at(0);
+  Parameter c1_b = net[0]->parameter_at(1);
+  Parameter f1_w = net[1]->parameter_at(0);
 
-  EXPECT_EQ(c1_w->size(), 9u);
-  EXPECT_EQ(c1_b->size(), 1u);
-  EXPECT_EQ(f1_w->size(), 2u);
+  EXPECT_EQ(c1_w.size(), 9u);
+  EXPECT_EQ(c1_b.size(), 1u);
+  EXPECT_EQ(f1_w.size(), 2u);
 
-  *c1_w = {0, 1, 2, 3, 4, 5, 6, 7, 8};
-  *c1_b = {1};
-  *f1_w = {1, 2};
+  c1_w.set_data(Tensor<float_t>{vec_t{0, 1, 2, 3, 4, 5, 6, 7, 8}});
+  c1_b.set_data(Tensor<float_t>{vec_t{1}});
+  f1_w.set_data(Tensor<float_t>{vec_t{1, 2}});
```

Comment:
Consider adding initializer-list assignment operator to ```Parameter``` to simplify these codes.